File: Desenvolvimento/server/src/ApplicationCore/Entities/Tramite.cs
using ApplicationCore.Entities.Identity;
using ApplicationCore.Interfaces.Base;
using System;
using System.Collections.Generic;
using System.Text;

namespace ApplicationCore.Entities
{
    public class Tramite : IBaseEntity<long>
    {
        public long Id { get; set; }
        public bool Tramitado { get; set; }
        public long AcaoId { get; set; }
        public long SetorId { get; set; }
        public long ProcessoId { get; set; }
        public string Observacao { get; set; }
        public long ResponsavelId { get; set; }
        public DateTime? DataTramite { get; set; }
        public bool EnviarEmailAutores { get; set; }
        public string EnviarEmailsPara { get; set; }

        public User Responsavel { get; set; }
        public FluxoAcao Acao { get; set; }
        public Setor Setor { get; set; }
        public Processo Processo { get; set; }

        public long SituacaoDoProcessoNoTramiteId { get; set; }
        public FluxoSituacao SituacaoDoProcessoNoTramite { get; set; } 

        public long SituacaoId { get; set; }
        public FluxoSituacao Situacao { get; set; }

        public virtual ICollection<TramiteArquivo> TramiteArquivos { get; set; }
        public virtual ICollection<TramiteChecklist> TramiteChecklists { get; set; }
    }
}

File: Desenvolvimento/client/src/@fuse/types/models/tramite.ts
import { Setor } from './setor';
import { User } from './user';
import { Processo } from './processo';
import { TramiteArquivo } from './tramite-arquivo';
import { TramiteChecklist } from './tramite-check-list';
import { FluxoAcao } from './fluxo-acao';
import { FluxoSituacao } from './fluxo-situacao';
import { BaseModel } from '../base-model';

export class Tramite extends BaseModel {
    constructor(
        public id: number,
        public tramitado?: boolean,
        public observacao?: string,
        public dataTramite?: Date,
        public enviarEmailAutores?: boolean,
        public enviarEmailsPara?: string,
        public acaoId?: number,
        public setorId?: number,
        public processoId?: number,
        public situacaoId?: number,
        public situacaoDoProcessoNoTramiteId?: number,
        public responsavelId?: number,
        public acao?: FluxoAcao,
        public setor?: Setor,
        public processo?: Processo,
        public situacao?: FluxoSituacao,
        public situacaoDoProcessoNoTramite?: FluxoSituacao,
        public responsavel?: User,
        public tramiteArquivos?: Array<TramiteArquivo>,
        public tramiteChecklists?: Array<TramiteChecklist>,
    ) { super(); }
}
